tests/stubs/devices/mockserver.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package devices
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/paypal/gatt"
+	"github.com/paypal/gatt/examples/option"
+
+	"github.com/kubeedge/kubeedge/tests/stubs/devices/services"
+)
+
+func main() {
+	d, err := gatt.NewDevice(option.DefaultServerOptions...)
+	if err != nil {
+		log.Fatalf("Failed to open device, err: %s", err)
+	}
+
+	// Register optional handlers.
+	d.Handle(
+		gatt.CentralConnected(func(c gatt.Central) { fmt.Println("Connect: ", c.ID()) }),
+		gatt.CentralDisconnected(func(c gatt.Central) { fmt.Println("Disconnect: ", c.ID()) }),
+	)
+
+	// A mandatory handler for monitoring device state.
+	onStateChanged := func(d gatt.Device, s gatt.State) {
+		fmt.Printf("State: %s\n", s)
+		switch s {
+		case gatt.StatePoweredOn:
+			// Setup GAP and GATT services for Linux implementation.
+			// OS X doesn't export the access of these services.
+			d.AddService(services.NewGapService("sensortagmock")) // no effect on OS X
+			d.AddService(services.NewGattService())               // no effect on OS X
+
+			// Creating a temperature reading service
+			s1 := services.NewTemperatureService()
+			d.AddService(s1)
+
+			// Advertise device name and service's UUIDs.
+			fmt.Print("Advertising device name and service UUID")
+			d.AdvertiseNameAndServices("sensortagmock", []gatt.UUID{s1.UUID()})
+
+			// Advertise as an OpenBeacon iBeacon
+			fmt.Print("Advertise as an OpenBeacon iBeacon")
+			d.AdvertiseIBeacon(gatt.MustParseUUID("AA6062F098CA42118EC4193EB73CCEB6"), 1, 2, -59)
+
+		default:
+		}
+	}
+
+	d.Init(onStateChanged)
+	select {}
```

Comment:
this will block forever. please add a stop channel which can receive kill signal and return ensuring a graceful shutdown.

Reply:
Updated